crossplane/README.md

```diff
@@ -0,0 +1,146 @@
+# Crossplane on Management Clusters
+
+Managed Crossplane offering is under development as per customers' requests.
+This document discusses all the required elements of Crossplane's deployment
+and the design decisions we make along the way.
+
+## Crossplane building blocks
+
+For ease of describing each proposal, we will divide a fully functional
+Crossplane deployment into three stages.
+
+### Stage 1 - Crossplane
+
+The first stage is Crossplane itself. This is not enough to manage any
+infrastructure yet.
+
+- Crossplane Operator
+- Crossplane RBAC Manager Operator
+- Custom Resource Definitions:
+  - `compositeresourcedefinitions.apiextensions.crossplane.io`
+  - `compositionrevisions.apiextensions.crossplane.io`
+  - `compositions.apiextensions.crossplane.io`
+  - `configurationrevisions.pkg.crossplane.io`
+  - `configurations.pkg.crossplane.io`
+  - `controllerconfigs.pkg.crossplane.io`
+  - `locks.pkg.crossplane.io`
+  - `providerrevisions.pkg.crossplane.io`
+  - `providers.pkg.crossplane.io`
+  - `storeconfigs.secrets.crossplane.io`
+- Other Kubernetes resources
+
+### Stage 2 - Crossplane Providers
+
+Infrastructure is managed by Providers. Creating a
+`providers.pkg.crossplane.io` resource will cause Crossplane operators to
+deploy provider operators and all necessary resources.
+
+- Provider-specific Operator
+- Provider-specific Custom Resource Definitions, usually
+`*.<provider-name>.crossplane.io`
+- Other Kubernetes resources
+
+### Stage 3 - Provider Configuration
+
+Just deploying Providers is not enough for them to be functional. They still
+need to be configured with passwords, secrets, access keys, API keys, and the
+like. Each Provider comes with its configuration CRD, usually
+`providerconfigs.<provider-name>.crossplane.io`.
+
+For example, the AWS provider will require a `ProviderConfig` similar to the
+one below to be able to perform reconciliation. `crossplane-system/aws-creds`
```

Comment:
Just FTR, the provider config can be / is different per provider, e.g. for AWS Jet:

```
apiVersion: aws.jet.crossplane.io/v1alpha1
kind: ProviderConfig
```

https://doc.crds.dev/github.com/crossplane-contrib/provider-jet-aws/aws.jet.crossplane.io/ProviderConfig/v1alpha1@v0.5.0-preview

Reply:
Yup, every provider's config kind is `ProviderConfig` but the `apiVersion`s differ.